### PR TITLE
Docs/docusaurus

### PR DESCRIPTION
**Pull request by jspizziri**
An overhaul of the existing documentation website with Docsaurus, reorganized existing documentation to be more digestable, fixed broken links.

### CODE REVIEW

**Comment by jspizziri** [comment]
@dcvz this PR would technically supersede this one https://github.com/doublesymmetry/react-native-track-player/pull/1487 if you want to move this direction.

For the record I think it's a massive improvement. You can run it locally and take a look. The reorg makes things easier to find and understand, we'll be able to easily integrate with Algolia for doc searching (which will be really nice), and a few other nice things.

Let me know your thoughts.

One thing I wanted to confirm is I think we can remove the `iOS Enable Swift Modules` instructions here https://react-native-track-player.js.org/install/

As I'm fairly certain I didn't do that last time I setup a project, but I could be forgetting.

File: docs/docusaurus.config.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+          {
+            title: 'Docs',
+            items: [
+              {
+                label: 'Intro',
+                to: '/docs/intro',
+              },
+              {
+                label: 'Getting Started',
+                to: '/docs/installation',
```
Comment (dcvz) [inline docs/docusaurus.config.js:76]:
```suggestion
                to: '/basics/installation',
```

File: docs/docusaurus.config.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                href: './docs/sponsors',
+              },
+              {
+                label: 'Core Team ✨',
+                href: './docs/core-team',
+              },
+            ],
+          },
+        ],
+        copyright: `Copyright © ${new Date().getFullYear()} DoubleSymmetry, Inc. Built with Docusaurus.`,
```
Comment (dcvz) [inline docs/docusaurus.config.js:119]:
```suggestion
        copyright: `Copyright © ${new Date().getFullYear()} Double Symmetry GmbH. Built with Docusaurus.`,
```

File: README.md [hunk 3]
```diff
@@ -163,5 +154,66 @@ start();
...
+Being a _champion_ makes it easy for us to help you. Which is what we all want!
+So how can you be a champion? [Sponsor the Project](https://github.com/sponsors/DoubleSymmetry) or _be willing to write some code_.
+
+**If _you're willing_** to write some code **_we're willing_** to:
+
+- Open a design discussion, give feedback, and approve something that works.
+- Provide guidance in the implementation journey.
+
+So, in a nutshell, let us know you're willing to do the work and ask for a little
+guidance, and watch the things you care about get done faster then anyone else.
```
Comment (dcvz) [inline README.md:191]:
```suggestion
guidance, and watch the things you care about get done faster than anyone else.
```

File: README.md [hunk 4]
```diff
@@ -163,5 +154,66 @@ start();
...
+The best help will be given to those who are willing to help themselves.
+
+###### You don't have experience you say? It's OK!
+
+You may be thinking that you can't help because you know nothing about native
+iOS or Android or maybe even React code. But we're willing to help teach you.
```
Comment (dcvz) [inline README.md:197]:
```suggestion
iOS or Android or maybe even React code. But we're willing to help guide you.
```
Don't want to overpromise that we can teach everything about a platform to someone, but we'll all do our best to help :)

**Review by dcvz — DISMISSED** [review]
huge PR ✨ I think this is a great improvement! Amazing job! :)
Left some small comments and some design notes below.

Also one question, will this automatically deploy via our Github CI jut like the current one?

**Some design notes:**
- Logo seems to not render correctly?
- We don't need both the logo and the title I'd say, just logo would work.
Screenshot 2022-04-22 at 22 59 39

**Comment by jspizziri** [comment]
@dcvz 

> Also one question, will this automatically deploy via our Github CI just like the current one?

So TBH I'm not sure how the current one is deploying. Not sure I have visibility into that. But here's what I do know

`cd docs && yarn build` will generate a static website located at `docs/build`. The files there just need to be pushed to the CDN.

File: docs/docusaurus.config.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+                href: './docs/sponsors',
+              },
+              {
+                label: 'Core Team ✨',
+                href: './docs/core-team',
+              },
+            ],
+          },
+        ],
+        copyright: `Copyright © ${new Date().getFullYear()} Double Symmetry GmbH Built with Docusaurus.`,
```
Comment (dcvz) [inline docs/docusaurus.config.js:119]:
```suggestion
        copyright: `Copyright © ${new Date().getFullYear()} Double Symmetry GmbH. Built with Docusaurus.`,
```

**Review by dcvz — APPROVED** [review]
LGTM 👍

**Comment by dcvz** [comment]
@jspizziri I'm actually also not sure how the current one is deploying 👯 
But I found this article about deploying docasaurus in Github Pages:
https://evantay.com/blog/docusaurus-gh-action/

**Comment by dcvz** [comment]
@jspizziri looks like Github automatically deploys your docs if you're using Jekkyl: https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site

**Comment by jspizziri** [comment]
@dcvz I just pushed some new workflows based on [the docusaurus documentation with gh-actions](https://docusaurus.io/docs/deployment#deploying-to-github-pages). I _think_ it will work. There's a test workflow in there for whenever someone makes a PR to `main`.

**Review by dcvz — APPROVED** [review]
Amazing work here @jspizziri. Let's merge it!

**Comment by jspizziri** [comment]
I'm going to check one more thing. There might be some special config for js.org. I think there needs to be a change file which I deleted.

**Comment by jspizziri** [comment]
@dcvz I had to push the CNAME file which got removed in the refactor. That's all that changed. I can merge once you hit the approve button again.

**Review by dcvz — APPROVED** [review]
Let's ship it!

**Pull request merged**